sqllogictest-bin/src/engines/postgres_extended.rs

```diff
@@ -113,115 +217,21 @@ impl sqllogictest::AsyncDB for PostgresExtended {
-                                    } else {
-                                        write!(output, "{}", value).unwrap();
-                                    }
-                                }
-                                None => {
-                                    write!(output, "NULL").unwrap();
-                                }
-                            }
-                        }
-                        Type::FLOAT8 => {
-                            let value: Option<f64> = row.get(idx);
-                            match value {
-                                Some(value) => {
-                                    if value == f64::INFINITY {
-                                        write!(output, "Infinity").unwrap();
-                                    } else if value == f64::NEG_INFINITY {
-                                        write!(output, "-Infinity").unwrap();
-                                    } else {
-                                        write!(output, "{}", value).unwrap();
-                                    }
-                                }
-                                None => {
-                                    write!(output, "NULL").unwrap();
-                                }
-                            }
-                        }
-                        Type::NUMERIC => {
-                            single_process!(row, output, idx, Decimal);
-                        }
-                        Type::TIMESTAMP => {
-                            single_process!(row, output, idx, NaiveDateTime);
-                        }
-                        Type::DATE => {
-                            single_process!(row, output, idx, NaiveDate);
-                        }
-                        Type::TIME => {
-                            single_process!(row, output, idx, NaiveTime);
-                        }
-                        Type::INT2_ARRAY => {
-                            array_process!(row, output, idx, i16);
-                        }
-                        Type::INT4_ARRAY => {
-                            array_process!(row, output, idx, i32);
-                        }
-                        Type::INT8_ARRAY => {
-                            array_process!(row, output, idx, i64);
-                        }
-                        Type::FLOAT4_ARRAY => {
-                            array_process!(row, output, idx, f32);
-                        }
-                        Type::FLOAT8_ARRAY => {
-                            array_process!(row, output, idx, f64);
-                        }
-                        Type::NUMERIC_ARRAY => {
-                            array_process!(row, output, idx, Decimal);
-                        }
-                        _ => {
-                            todo!("Don't support {} type now.", column.type_().name())
-                        }
-                    }
+                    generate_process!(column,row, output,idx,
```

Comment:
Why to use macro here? How is it simplier than before?

Reply:
reduce duplication like
```
Type::NUMERIC => {
    single_process!(row, output, idx, Decimal);
}
Type::TIMESTAMP => {
    single_process!(row, output, idx, NaiveDateTime);
}
Type::DATE => {
    single_process!(row, output, idx, NaiveDate);
}
...

```

Reply:
IMO such level of duplication is fine. Macro seems an overkill and hurts readability.